st2client/st2client/config_parser.py

```diff
@@ -138,6 +139,20 @@ def parse(self):
             # Config doesn't exist, return the default values
             return CONFIG_DEFAULT_VALUES
 
+        config_dir_path = os.path.dirname(self.config_file_path)
+
+        if bool(os.stat(config_dir_path).st_mode ^ 0o770):
+            warnings.warn('Setting StackStorm config directory permissions '
```

Comment:
Instead of using `warnings` module we should probably use ``logger.warn`` since we also use logging in other places for such things.

This way it uses the existing framework and setting log level, etc works as expected (e.g. user wants to suppress warnings and only see messages with level ERROR and above).